src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private Entry<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Entry[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            if (table[0] == null) {
+                table[0] = new Entry<>(null, value);
+            } else {
+                Entry<K, V> current = table[0];
+                while (current != null) {
+                    if (current.key == null) {
+                        current.value = value;
+                        return;
+                    }
+                    current = current.next;
+                }
+                Entry<K, V> entry = new Entry<>(null, value);
+                entry.next = table[0];
+                table[0] = entry;
+            }
+        } else {
+            int index = hash(key);
+            Entry<K, V> entry = new Entry<>(key, value);
 
+            if (table[index] == null) {
+                table[index] = entry;
+            } else {
+                Entry<K, V> current = table[index];
+                while (current != null) {
+                    if (current.key != null && current.key.equals(key)) {
+                        current.value = value;
+                        return;
+                    }
+                    current = current.next;
+                }
+                entry.next = table[index];
+                table[index] = entry;
+            }
+        }
+        size++;
+        if ((double) size / table.length >= LOAD_FACTOR) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = hash(key);
+        Entry<K, V> current = table[index];
+
+        while (current != null) {
+            if (current.key == null && key == null
```

Comment:
this condition may be simplified ( == will give true only in case it's same object or both are nulls)
```suggestion
            if (current.key == key
```